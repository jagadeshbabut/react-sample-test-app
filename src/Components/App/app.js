import React, {Component} from 'react';
import './app.css';
import {BrowserRouter as Router, Route} from 'react-router-dom';
import Signup from './../Signup/signup';
import Signin from './../Signin/signin';
import Dashboard from './../Dashboard/dashboard';
import Todo from './../Todo/todo';

class App extends Component {
  render() {
    return (
      <Router>
        <div>
          <Route exact path="/" component={Signup}/>
          <Route path="/signin" component={Signin}/>
          <Route path="/dashboard" component={Dashboard}/>
          <Route path="/todos" component={Todo}/>
        </div>
      </Router>
    );
  }
}

export default App;
