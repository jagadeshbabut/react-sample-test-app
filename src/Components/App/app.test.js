import React from 'react';
import App from './app';
import {shallow} from 'enzyme';

describe('App Component Testing', () => {
  it('App Renders without crashing', () => {
    shallow(<App/>);
  });

  it('App component has not changed', () => {
    const wrapper = shallow(<App/>);
    expect(wrapper).toMatchSnapshot()
  });

});