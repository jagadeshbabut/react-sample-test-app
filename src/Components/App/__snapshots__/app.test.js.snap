// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Testing App component has not changed 1`] = `
<BrowserRouter>
  <div>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      path="/signin"
    />
    <Route
      component={[Function]}
      path="/dashboard"
    />
    <Route
      component={[Function]}
      path="/todos"
    />
  </div>
</BrowserRouter>
`;
